packages/checkout/sdk/src/types/smartCheckout.ts

```diff
@@ -38,6 +43,21 @@ export type BuyResultFailed = {
   smartCheckoutResult: SmartCheckoutSufficient
 };
 
+/**
+ * Represents the result of {@link Checkout.buy}
+ * @property {CheckoutStatus.FULFILLMENTS_UNSETTLED} status
+ * @property {SmartCheckoutSufficient} smartCheckoutResult
+ * @property {SendTransactionResult[]} transactions
+ */
+export type BuyResultFulfillmentsUnsettled = {
+  /** The status to indicate success */
+  status: CheckoutStatus.FULFILLMENTS_UNSETTLED,
+  /** The sufficient result of smart checkout */
+  smartCheckoutResult: SmartCheckoutSufficient,
+  /** Array of transaction results */
+  transactions: SendTransactionResult[],
```

Comment:
`SendTransactionResult` is the type for the return of `sendTransaction`. These param/result types are specific for that sdk functions return type. I think because of this that type shouldn't be used with the `BuyResultFulfillmentsUnsettled`

Thinking about this and imo its probably fine to have this an array of `TransactionResponse` types as I dont think we would need any additional info in each of those transactions, but if we think its better for extensibility to have this on a new type then we should create a new one instead of using `SendTransactionResult` for the reason above.

Main downside with having this as a new object type instead of `TransactionResponse` is thats another layer (eg `transactions[i].transactionResponse.wait()` instead of `transactions[i].wait()`

Thoughts?

Reply:
Initial thought here was to not expose when possible etherjs specific types. Beside the web3 provider we use etherjs in other exposed types so perhaps returning `TransactionResponse` is just fine. Given that we have a `BuyResultFulfillmentsUnsettled` specific type, it shouldn't be too hard in the future to extend this response.

Reply:
I've updated this to return `TransactionResponse`. I think using more direct structure makes sense to me and it's not tied to the sendTransaction result as you mentioned @imx-mikhala